setup.sh

```diff
@@ -110,6 +110,100 @@ function check_dependencies()
   fi
 }
 
+# This function checks and installs kernel build dependencies.
+# It supports three Linux distributions: Debian, Arch Linux, and Fedora.
+#
+# Returns:
+# return 0: All necessary dependencies are already installed or installation is successful.
+# return 1 (EPERM): Unsupported OS type.
+function check_and_install_kernel_build_dependencies()
+{
+  local package_list
+  local cmd
+  local distro
+  local kernel_build_deps_file
+  local installed
+
+  distro=$(detect_distro '/')
+
+  kernel_build_deps_file="${DOCUMENTATION}/dependencies/kernel_build/${distro}.dependencies"
+
+  case "$distro" in
+    arch*)
+      while IFS='' read -r package; do
+        installed=$(pacman -Ql "$package" &> /dev/null)
+        [[ "$?" != 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="pacman -Sy --noconfirm ${package_list}"
+      ;;
+    debian*)
+      while IFS='' read -r package; do
+        installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
+        [[ "$installed" -eq 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="apt install -y ${package_list}"
+      ;;
+    fedora*)
+      while IFS='' read -r package; do
+        installed=$(rpm -q "$package" &> /dev/null)
+        [[ "$?" -ne 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="dnf install -y ${package_list}"
+      ;;
+    *)
+      complain "Unsupported OS type: ${distro}"
+      return 1 # EPERM
+      ;;
+  esac
```

Comment:
I see that you took part of the `check_dependencies` as inspiration for this block that I am highlighting. Maybe we can extract a function to reduce duplication that accepts a distro and a dependencies file as arguments (in this case, the line `distro=$(detect_distro '/')` wouldn't be included in the function), and returns the `package_list`. The con of  this, is that we would still have a (smaller) switch-case in both `check_and_install_kernel_build_dependencies` and `check_dependencies` to build the correct `cmd` value.

This is just a rough idea, nevertheless, I think somehow extracting this duplicated switch-case (if-elif-else in `check_dependencies`) is cleaner.